### PR TITLE
internal/extsvc: Flexible check for parts in reponame

### PR DESCRIPTION
**Pull request by indradhanush**

### CODE REVIEW

File: internal/extsvc/codehost.go [hunk 1]
```diff
@@ -74,7 +74,12 @@ func NormalizeBaseURL(baseURL *url.URL) *url.URL {
 // matching codehost from the given list.
 func CodeHostOf(name api.RepoName, codehosts ...*CodeHost) *CodeHost {
 	repoNameParts := strings.Split(string(name), "/")
```
Comment (tsenart) [inline internal/extsvc/codehost.go:76]: I think this should do the trick. We only need 2 parts.
```suggestion
	repoNameParts := strings.SplitN(string(name), "/", 2)
```

Comment (indradhanush) [reply internal/extsvc/codehost.go:76]: Wouldn't that mean repo name of the format - `github.com/foo` will pass as valid repo names and then throw the same error that we get right now? :thinking:

Comment (tsenart) [reply internal/extsvc/codehost.go:76]: The problem we have right now is that the hostname matches but shouldn't, not the path.

Comment (indradhanush) [reply internal/extsvc/codehost.go]: Fair enough. Fixed and updated docs.

**Review by tsenart — APPROVED** [review]

**Pull request merged**